build_projects/dotnet-host-build/MsiTargets.cs

```diff
@@ -162,6 +162,7 @@ public static BuildTargetResult GenerateDotnetHostFxrMsi(BuildTargetContext c)
             var inputDir = c.BuildContext.Get<string>("HostFxrPublishRoot");
             var wixObjRoot = Path.Combine(Dirs.Output, "obj", "wix", "hostfxr");
             var hostFxrBrandName = $"'{Monikers.GetHostFxrBrandName(c)}'";
+			var upgradeCode = Utils.GenerateGuidFromName(HostFxrMsi).ToString().ToUpper();
```

Comment:
Can you also add a comment about this change here so that someone reading it in future would know why this is required?